src/main/java/io/deepstream/PresenceHandler.java

```diff
@@ -0,0 +1,140 @@
+package io.deepstream;
+
+
+import com.google.j2objc.annotations.ObjectiveCName;
+
+import java.util.*;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+public class PresenceHandler {
+
+    private final int subscriptionTimeout;
+    private final UtilEmitter emitter;
+    private final DeepstreamConfig deepstreamConfig;
+    private final IConnection connection;
+    private final DeepstreamClientAbstract client;
+    private final UtilAckTimeoutRegistry ackTimeoutRegistry;
+    private final UtilSingleNotifier notifier;
+
+    PresenceHandler(DeepstreamConfig deepstreamConfig, final IConnection connection, DeepstreamClientAbstract client) {
+        this.subscriptionTimeout = deepstreamConfig.getSubscriptionTimeout();
+        this.connection = connection;
+        this.client = client;
+        this.emitter = new UtilEmitter();
+        this.deepstreamConfig = deepstreamConfig;
+        this.ackTimeoutRegistry = client.getAckTimeoutRegistry();
+        this.notifier = new UtilSingleNotifier(client, connection, Topic.PRESENCE, Actions.QUERY, subscriptionTimeout);
+
+        new UtilResubscribeNotifier(this.client, new UtilResubscribeNotifier.UtilResubscribeListener() {
```

Comment:
because we wouldn't want the interface on the public api, you would need to have a private class that implements it and then use that class. Guessing this makes more sense for a single api!